darwin/importer/importer.py

```diff
@@ -205,20 +210,41 @@ def import_annotations(
     else:
         remote_classes = build_main_annotations_lookup_table(team_classes)
 
-    # Need to re parse the files since we didn't save the annotations in memory
-    for local_path in set(local_file.path for local_file in local_files):
-        parsed_files = importer(local_path)
-        if type(parsed_files) is not list:
-            parsed_files = [parsed_files]
-        # remove files missing on the server
-        missing_files = [missing_file.full_path for missing_file in local_files_missing_remotely]
-        parsed_files = [parsed_file for parsed_file in parsed_files if parsed_file.full_path not in missing_files]
-        for parsed_file in track(parsed_files):
+    with ThreadPoolExecutor(max_workers=max_workers) as executor:
+
+        def get_parsed_files(local_path):
+            parsed_files = importer(local_path)
+            if type(parsed_files) is not list:
+                parsed_files = [parsed_files]
+            # remove files missing on the server
+            missing_files = [missing_file.full_path for missing_file in local_files_missing_remotely]
+            parsed_files = [parsed_file for parsed_file in parsed_files if parsed_file.full_path not in missing_files]
+            return parsed_files
+
+        def import_annotation(parsed_file):
             image_id = remote_files[parsed_file.full_path]
             _import_annotations(
```

Comment:
How is this going to behave if the `post` in the function fails and we raise?

Reply:
I don't know, I just refactor the code that was there